### ISSUE TITLE
ClientError: An error occurred (InvalidArgument) when calling the ListObjects operation: Unknown

### ISSUE THREAD

**Issue by Klaudioz** [mobarski/ask-my-pdf#3]
I got the following error:

```
ClientError: An error occurred (InvalidArgument) when calling the ListObjects operation: Unknown
Traceback:
File "/usr/local/lib/python3.10/site-packages/streamlit/runtime/scriptrunner/script_runner.py", line 565, in _run_script
    exec(code, module.__dict__)
File "/Users/canalescl/personal/replit/ask-my-pdf/src/gui.py", line 248, in <module>
    ui_pdf_file()
File "/Users/canalescl/personal/replit/ask-my-pdf/src/gui.py", line 91, in ui_pdf_file
    filenames += ss['storage'].list()
File "/Users/canalescl/personal/replit/ask-my-pdf/src/storage.py", line 46, in list
    return [self.decode(name) for name in self._list()]
File "/Users/canalescl/personal/replit/ask-my-pdf/src/storage.py", line 184, in _list
    resp = self.s3.list_objects(
File "/usr/local/lib/python3.10/site-packages/botocore/client.py", line 530, in _api_call
    return self._make_api_call(operation_name, kwargs)
File "/usr/local/lib/python3.10/site-packages/botocore/client.py", line 960, in _make_api_call
    raise error_class(parsed_response, operation_name)
```

A new folder, cloned from zero, after "Enter your OpenAI API key" and click Enter

![image](https://user-images.githubusercontent.com/632625/220764644-38688085-729f-4131-b1de-6e4793dd7e16.png)

**Comment by mobarski** [comment]
I'll fix this tomorrow - S3 usage must be disabled when no S3 credentials are provided.

**Comment by mobarski** [comment]
Fixed in 0.4.1.
DictStorage is used as the default storage.
When S3 storage is used and no S3 credentials are present in the environment variables exception will be raised during storage initialisation.
